test/cql-pytest/test_materialized_view.py

```diff
@@ -992,3 +992,86 @@ def test_base_clustering_prefix_deletion(cql, test_keyspace):
             # After the deletion, all data should be gone from both base and view
             assert [] == list(cql.execute(f"SELECT c2 FROM {table}"))
             assert [] == list(cql.execute(f"SELECT c2 FROM {mv}"))
+
+# Test deleting an entire base partition, where there is a view with the same or
+# different partition key.  When the view has the same partition key as base,
+# the partition deletion can be done on the base in one update of partition
+# tombstone.  In other cases, a tombstone is generated for each row that
+# corresponds to the deleted base partition.
+def test_base_partition_deletion_with_same_view_partition_key(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, 'p int, c int, v int, primary key (p,c)') as table:
+        # Insert into two base partitions. We will delete one of them
+        v = 42
+        insert = cql.prepare(f'INSERT INTO {table} (p,c,v) VALUES (?,?,{v})')
+        # Create multiple views with different primary keys.
+        # The first view has the same partition key as the base, so the entire partition will be deleted on the view as well.
+        # The other views have different partition key than the base, so they will have individual rows removed.
+        with new_materialized_view(cql, table, '*', 'p,c', 'p is not null and c is not null', extra='with synchronous_updates = true') as mv1, \
```

Comment:
The with_synchronous_updates=true will not work on Cassandra, so you won't be able to run this test on Cassandra. I suggest you can just drop it - on a single-node setup the view updates are synchronous in any case.

Reply:
done